### PR TITLE
Improved auto drive

### PR DESCRIPTION
**Pull request by xM4ddy**
Split drive to WP and wander, made is so that you can change speed and driving style mid drive, and changed the driving styles to the ones from #190.

### CODE REVIEW

**Comment by xM4ddy** [comment]
I know I know

File: BigBaseV2/src/views/vehicle/view_vehicle.cpp [hunk 1]
```diff
@@ -58,24 +58,31 @@ namespace big
...
+
+			QUEUE_JOB_BEGIN_CLAUSE()
+			{
+				g->vehicle.auto_drive_to_waypoint = false;
+				g->vehicle.auto_drive_wander = false;
+				VEHICLE::SET_VEHICLE_FORWARD_SPEED(self::veh, 0);
+				TASK::CLEAR_VEHICLE_TASKS_(self::veh);
+				TASK::CLEAR_PED_TASKS(self::ped);
+			}
+			QUEUE_JOB_END_CLAUSE
```
Comment (Yimura) [inline BigBaseV2/src/views/vehicle/view_vehicle.cpp]: Wrapped inside a queue job? The button component callback automatically wraps its callbacks in fiber jobs, no need for an additional queue job.

File: BigBaseV2/src/backend/looped/vehicle/auto_drive_wander.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+				if (!changing_driving_styles)
+				{
+					g_notification_service->push_warning("Starting Wondering", "Start driving or leave car to take back control.");
+				}
+			}
+		}
+
+		if (wandering && ran_once)
+		{
+			if (PAD::IS_CONTROL_PRESSED(0, 63) || PAD::IS_CONTROL_PRESSED(0, 64) || PAD::IS_CONTROL_PRESSED(0, 71) || PAD::IS_CONTROL_PRESSED(0, 72) || PAD::IS_CONTROL_PRESSED(0, 75) || PAD::IS_CONTROL_PRESSED(0, 76))
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/vehicle/auto_drive_wander.cpp]: Replace controller input `int` with its enum equivalent from `gta/enums.hpp`, @karifeld went through a great deal of trouble replacing all the controller values inside YimMenu to the gta enums.

File: BigBaseV2/src/backend/looped/vehicle/auto_drive_to_waypoint.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+				g->vehicle.auto_drive_to_waypoint = false;
+
+				driving_to_wp = true;
+			}
+		}
+
+		if (driving_to_wp)
+		{
+			if (!blip::get_blip_location(location, (int)BlipIcons::Waypoint) || PAD::IS_CONTROL_PRESSED(0, 63) || PAD::IS_CONTROL_PRESSED(0, 64) || PAD::IS_CONTROL_PRESSED(0, 71) || PAD::IS_CONTROL_PRESSED(0, 72) || PAD::IS_CONTROL_PRESSED(0, 75) || PAD::IS_CONTROL_PRESSED(0, 76))
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/vehicle/auto_drive_to_waypoint.cpp]: Replace controller input int with its enum equivalent from gta/enums.hpp, @karifeld went through a great deal of trouble replacing all the controller values inside YimMenu to the gta enums.

**Review by Yimura — CHANGES_REQUESTED** [review]

File: BigBaseV2/src/backend/looped/vehicle/auto_drive_wander.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
+#include "backend/looped/looped.hpp"
+#include "natives.hpp"
+#include "util/blip.hpp"
+#include "util/entity.hpp"
+#include "util/vehicle.hpp"
+#include "gta/enums.hpp"
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/vehicle/auto_drive_wander.cpp]: includes alphabetically

File: BigBaseV2/src/backend/looped/vehicle/auto_drive_to_waypoint.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
+#include "backend/looped/looped.hpp"
+#include "natives.hpp"
+#include "util/blip.hpp"
+#include "util/entity.hpp"
+#include "util/vehicle.hpp"
+#include "gta/enums.hpp"
```
Comment (Yimura) [inline BigBaseV2/src/backend/looped/vehicle/auto_drive_to_waypoint.cpp]: includes alphabetically

**Review by Yimura — CHANGES_REQUESTED** [review]

**Review by Yimura — APPROVED** [review]

**Pull request merged**